src/app/core/common-components/entity-form/entity-form/entity-form.component.ts

```diff
@@ -78,6 +81,31 @@ export class EntityFormComponent<T extends Entity = Entity>
       this.initialFormValues = this.form.getRawValue();
       this.disableForLockedEntity();
     }
+
+    if (this.form && this.entity) {
+      this.entityFormService.disableReadOnlyFormControls(
+        this.form,
+        this.entity,
+      );
+    }
+  }
+
+  private applyFormFieldPermissions() {
+    if (this.fieldGroups) {
+      this.fieldGroups = this.entityFormService.filterFieldGroupsByPermissions(
+        this.fieldGroups,
+        this.entity,
+      );
+    }
+
+    if (this.form) {
+      this.form.statusChanges.pipe(untilDestroyed(this)).subscribe((_) => {
```

Comment:
I guess this does not need to be run for every status change (e.g. not when the status changes to `disabled`).

Reply:
I will add a check for disabled state.